### ISSUE TITLE
right-clicking on terminal OUTPUT unexpectedly pastes to command line

### ISSUE THREAD

**Issue by carehart** [microsoft/vscode#78295]
Folks, I wanted to point out an issue I tripped over. Perhaps it may affect others. It has to do with the expected behavior when one does right-click in terminal, causing text to be pasted. For someone coming from experience with the Windows command prompt, the behavior is not **quite** the same, as it happens if you right-click on the OUTPUT rather than only if you right-click on the command prompt itself. 

That may surprise them or even cause trouble, as I will explain. And while there may seem already a solution to it, there's an issue with that as well. 

I offer below the issue, then some discussion, then a suggested resolution, though I'll understand if no change is made. I welcome feedback from anyone interested in the matter.

**Issue**

With the Windows command prompt, a right-click only pastes text if you are ON the command prompt itself, where it's awaiting input. If you right-click on the text output of previous commands, it does NOT do the paste.

But with the VSCode terminal ui (and the default of copyPaste for the preferences setting of features>terminal>right-click behavior), if one does a right-click ANYWHERE in the terminal window (including on command output) that will indeed do the paste to the command prompt.

**Discussion**

This threw me off the first time I was using the terminal feature, as a right-click within an IDE will bring up a context menu. I thought it may help me to select all the output, for instance. 

Instead, it pasted some text I had in my clipboard (having nothing to do with what I was working with in vscode), which included line-end characters, and the terminal started throwing errors. Fortunately nothing bad happened, but it could have if I didn't realize it was going to happen.

Again, the key is that I was NOT right-clicking on the terminal INPUT area (the actual command prompt). I was right-clicking on the output, for reason I said above. 

** Existing alternative **

Now, I do see (on digging into this) that the preference setting for this terminal right-click behavior DOES ALSO offer an alternative option called "default" (which not set by default on a new install), and that DOES indeed show a context menu if one does a right-click. Fair enough.

And someone may propose I switch to THAT, as they may LIKE how they don't need to be "right on the input" field of Terminal for the paste to happen. I'm just saying first that it's unexpected behavior for one coming from the Windows command prompt. :-)

But second, and as interesting, switching to that "default" option still does NOT behave like the Windows command prompt when it comes to pasting: again if you right-click on the *input area* of the Windows command prompt, a right-click DOES do a paste. But this does NOT happen in VSCode, with the "default" setting for the terminal right-click behavior.

**Suggested resolution**

So it seems that for consistency sake at least, the "default" setting should at least be changed to do the paste for a right-click on the input area, to better model the Command prompt behavior, for those who may change from the default CopyPaste behavior. 

Some might also wish that the (real default) copyPaste setting would also ONLY paste if you were on the terminal input area. But I can understand why folks already used to it might regard this as a feature, in they don't have to be so precise about pasting on the terminal input line, and so wouldn't want it changed.

Hope that's seen as helpful. I'm not being critical. Just pointing out the inconsistency. I realize all choices like this are hard and involve juggling many priorities and constituencies. :-)

VS Code version: Code 1.36.1 (2213894ea0415ee8c85c5eea0d0ff81ecc191529, 2019-07-08T22:59:35.033Z)
OS version: Windows_NT x64 10.0.18362

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz (8 x 2808)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>oop_rasterization: disabled_off<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_deferred_display_list: disabled_off<br>skia_renderer: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.89GB (3.57GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (5)</summary>

Extension|Author (truncated)|Version
---|---|---
auto-close-tag|for|0.5.6
vscode-cfml|Kam|0.5.3
vscode-docker|ms-|0.7.0
cpptools|ms-|0.24.1
csharp|ms-|1.21.0

</details>

**Comment by Tyriar** [comment]
@carehart the right click paste is indeed modeled after conhost.

> With the Windows command prompt, a right-click only pastes text if you are ON the command prompt itself, where it's awaiting input. If you right-click on the text output of previous commands, it does NOT do the paste.

I don't see this, in conhost right clicking on _any_ line to pastes from the clipboard. @DHowett-MSFT is there anything special about the line that is right clicked in conhost/windows terminal to paste?

**Comment by DHowett-MSFT** [comment]
Your assertion is correct. Right-clicking ANYWHERE in conhost enacts a paste operation.

**Comment by Tyriar** [comment]
Closing as designed then, @carehart let me know if I'm misunderstanding something.

**Comment by carehart** [comment]
So very odd. I would not have filed all that if I'd not seen it happening. But today I can confirm that I don't see it happening (I find that right-clicking on any command prompt window, anywhere, does indeed paste to the input area.) I should add that this is only if quickedit is on (in that command-prompt window's properties), but you all must have that on also. I do. 

I can also confirm that I only noticed today that there is the option for the terminal in vscode to be either a ps or cmd prompt. I'm almost positive I was viewing a cmd one. I have several machines and can't recall now which one I was working on when I saw and was tempted to write.

So for now I will chalk it up to geezery. Apologies for the seeming false alarm. If I am able, I may well delete this in coming days so as not to trouble anyone who might look at it in closed issues.

VS Code is great. Thanks to all for the efforts.
